libgnucash/engine/Account.cpp

```diff
@@ -1151,6 +1151,19 @@ gnc_account_foreach_split (const Account *acc, std::function<void(Split*)> func,
         std::for_each(splits.begin(), splits.end(), func);
 }
 
+void
+gnc_account_foreach_until_date (const Account *acc, time64 end_date, std::function<void(Split*)> f)
+{
+    auto splits{GET_PRIVATE(acc)->splits};
+
+    auto is_before = [](time64 end_date, auto s) -> bool
+    { return (xaccTransGetDate(xaccSplitGetParent (s)) <= end_date); };
```

Comment:
That makes it "is_before_or_on" unless `end_date` is after 10:59:59Z on the day.